src/CommandingOfficers/CommanderAve.java

```diff
@@ -0,0 +1,646 @@
+      }
+    }
+
+    addCommanderAbility(new NixAbility(this));
+    addCommanderAbility(new GlacioAbility(this));
+    addCommanderAbility(new OblidoAbility(this));
+    snowifier = new CitySnowifier(this);
+    GameEventListener.registerEventListener(snowifier);
+  }
+
+  @Override
+  public GameEventQueue initTurn(GameMap gameMap)
+  {
+    GameEventQueue returnEvents = super.initTurn(gameMap);
+
+    // Initialize our snow tracker if needed.
+    if( null == snowMap )
+    {
+      // Start by initializing owned properties to SNOW_THRESHOLD
+      snowMap = new int[gameMap.mapWidth][gameMap.mapHeight];
+      addSnow(SNOW_THRESHOLD, gameMap, returnEvents);
+    }
+
+    // Add snow to all buildings we own.
+    addSnow(SNOW_PER_TURN, gameMap, returnEvents); // Enough to spread a bit each way.
+    relevelSnow(gameMap, returnEvents); // Enough to spread a bit each way.
+
+    if(snowLoggingEnabled) log("[Commander Ave.initTurn] Snow Map: \n" + getSnowMapAsString());
+    return returnEvents;
+  }
+
+  /** Ave's units take less cover from forests. */
+  @Override
+  public void applyCombatModifiers(BattleParams params, boolean amITheAttacker)
+  {
+    if( params.defender.CO == this // We are defending, in a FOREST
+        && (params.combatRef.gameMap.getEnvironment(params.defender.x, params.defender.y).terrainType == TerrainType.FOREST)
+        && (params.defender.model.isLandUnit()))  // And our unit is actually on the ground.
+    {
+      params.terrainDefense--;
+    }
+  }
+
+  private boolean snowLoggingEnabled = false;
+  public void log(boolean enable)
+  {
+    snowLoggingEnabled = enable;
+  }
+  private void log(String message)
+  {
+    if(snowLoggingEnabled) System.out.println(message);
+  }
+
+  /**
+   * Dump extra snow on Ave's properties and then spread it around.
+   * outEvents will be populated with any MapChangeEvents that result.
+   */
+  private void addSnow(double amount, GameMap gameMap, GameEventQueue outEvents)
+  {
+    log("============ Adding snow ========================================");
+    ArrayList<MapChangeEvent.EnvironmentAssignment> tiles = new ArrayList<MapChangeEvent.EnvironmentAssignment>();
+    for( XYCoord xyc : ownedProperties )
+    {
+      // Boost the amount of snow power this property has.
+      snowMap[xyc.xCoord][xyc.yCoord] += amount;
+      if( gameMap.getEnvironment(xyc).weatherType != Weathers.SNOW )
+      {
+        Environment envi = Environment.getTile(gameMap.getEnvironment(xyc).terrainType, Weathers.SNOW);
+        tiles.add(new MapChangeEvent.EnvironmentAssignment(xyc, envi, 2));
+      }
+    }
+    if( !tiles.isEmpty())
+    {
+      GameEvent event = new MapChangeEvent(tiles);
+      outEvents.add(event);
+    }
+  }
+
+  /**
+   * Spread snow around.
+   * outEvents will be populated with any MapChangeEvents that result.
+   */
+  private void relevelSnow(GameMap gameMap, GameEventQueue outEvents)
```

Comment:
This is admittedly a pretty cool algorithm, but I think maybe it's a little overly-complex.
I'm not a huge fan of having functions that won't comfortably fit on a monitor, especially if it's a whole bunch of actual complex calculations. (sidenote: for this reason, I was planning to make a "meta AI" that does flow-of-control stuff, so other AIs can just define whatever phases they want.)

Outside of making the logic less complex, would you consider at least just breaking it up into sub-functions so it's easier to follow?

Reply:
> EDIT: I feel like this is especially relevant since her powers no longer use the snow map at all.

Sounds like the real solution is to change her powers to rely on the snow map again, eh?

Yeah, I agree this is more complex than it really needs to be at this point. However, it also works very well without issues, so in the spirit of "don't fix what ain't broken" I just wasn't gonna mess with it unless I found out it was broken.
If you think it's worth finding a simpler alternative despite that we can though.

I'm not a fan of using `log(turnsSinceCapture)` - it levels off too quickly, meaning farther spaces would take basically forever. It also doesn't account for the current behavior of allowing existing snow to overflow into new spaces so they fill faster (though this is probably not all that important).
Another simple approach might be something like:
```
def isSnow(dist, turns):
...   needed = 2*dist-1
...   if( needed <= turns ): return True
...   else: return False
```
This would fill in distance 1 turn, and an additional space every two subsequent turns.

Alternatively, do a Game of Life approach where tiles are filled in once they have neighbors filled in (maybe for a couple of turns), though this works out almost the same as the previous thing.

Reply:
I guess that's fair.

I think I probably want to poke around and test a bit to be sure, but from a code-inspection standpoint I guess I'm done with this review.